New package: nvidia-open-dkms-580.142.

JkktBkkt opened the pull request:
~~Important: PR has been set to WIP as the template requires additional review, for cleanup and tie-in with main nvidia package.~~
Unset, current version should be a bit cleaner, not requiring manual removal of -dkms before re-installing updated -open-dkms on every update. 

~~Here's a permalink to the commit that doesn't change the `nvidia` template if you want to install or upgrade before this can land and in case more changes are requested by maintainers: https://github.com/void-linux/void-packages/pull/54593/commits/78cc73d0ce554abeeedc7065e1cb641b23250832~~
^ current version doesn't change `nvidia` template either, but see there for previous update notes.

See https://github.com/void-linux/void-packages/pull/54593#issuecomment-3411057607 and below to track that progress

#### Testing the changes
- I tested the changes in this PR: **YES**

<!--
#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**
-->

#### Local build testing
- I built this PR locally for my native architecture, x86_64-glibc 

**Comments**
Most recent versions I test on 6.18
No longer WIP since this release is out of [beta](https://us.download.nvidia.com/XFree86/Linux-x86_64/570.124.04/README/kernel_open.html) and is now the recommended driver for supported devices, which are Turing and newer, so 16-series and newer of the GeForce lineup.
Closes #51384 
Perhaps package info should mention which GPUs are supported

Thanks to mike7d7 for original template and PR #51538

mike7d7 committed:
New package: nvidia-open-dkms-580.142.

Co-authored-by: Miguel <migue07mx@protonmail.com>

JkktBkkt commented:
Being not really comfortable with git cli I tried to preserve mike's contribution (and failed miserably as can be seen below)
Just double-checked by cloning the pr as described in contributing.md and re-ran per-stage `./xbps-src {stage} -f -K nvidia-open-dkms` to verify no issues preset.
Understanding that this smells of blind editing with no testing done, here's my `xbps-query nvidia-open-dkms`
```
architecture: x86_64
changelog: https://github.com/NVIDIA/open-gpu-kernel-modules/blob/main/CHANGELOG.md
filename-sha256: 8380e0918aa0db31a732331d378116859938b07afa1f8fecfe32662e313ff607
filename-size: 14MB
homepage: https://github.com/NVIDIA/open-gpu-kernel-modules
install-date: 2025-03-02 21:18 MSK
install-script: 950B
installed_size: 100MB
license: GPL-2.0-only, MIT
maintainer: mike7d7 <mike7d7@proton.me>
metafile-sha256: f57e061a8bdbdda880176a90028ef935820d92799c29377065e3f5017c17b046
pkgname: nvidia-open-dkms
pkgver: nvidia-open-dkms-570.124.04_1
remove-script: 947B
replaces:
        nvidia-dkms>=0
repository: /home/jkkt/Projects/void-packages/hostdir/binpkgs/nvidia-open
run_depends:
        dkms>=0
        nvidia>=0
short_desc: NVIDIA Linux open GPU kernel module source
state: installed
```
And dkms make log:
[make.log](https://github.com/user-attachments/files/19106361/make.log)

JkktBkkt commented:
Oh, and also this makes `nvidia` package (in current state) report not satisfied dependency in form of nvidia-dkms, as that's replaced by this driver. So far I'm not up to speed with xbps-src and templates enough to understand how exactly that should be fixed.

JkktBkkt commented:
@classabbyamp can re-build be approved?

JkktBkkt commented:
@abenson If I may bother you, since you're the maintainer of the proprietary driver package, what would be the appropriate way for dealing with this conflicting files issue? Especially considering that not all cards supporting 570.124.04 can run the open-source version of it. 
Would adding `conflicts="nvidia470-dkms>=0 nvidia390-dkms>=0"` be enough? 
Would, perhaps, transferring this into the main package while defaulting to open for newer versions and leaving the proprietary version symlinked (like 390/470 versions are) be preferred? (Although then the question of these having different licenses arises too)

fvalasiad commented:
Hello,

While I do not have anything more to add as of now, I'd like to answer your question about the `conflicts` situation.

No, it would not be enough, the nvidia470 & nvidia390 respectively support older nvidia cards, but there exist newer nvidia cards that are still not supported by the open driver, for example the GTX 10 series.

about how that could go down well, personally I'd look into having an equivalent proprietary package(if there ain't one already) and have them conflict each other, with the open one picked as a default while adding an appropriate section in the void handbook for those cards in between, but that's me so take everything with a spoonful of salt.

Definitely ain't a simple change though, so practice patience, and just keep bumping those versions, eventually attention will be brought to this topic out of necessity.

JkktBkkt commented:
Hi, thanks for your reply. 
Regarding the similar proprietary package that would conflict this one — to my knowledge, that's partially how "replaces" here operates, and is about as much as I can do without messing with abenson's `nvidia` template. 
Adding this template into conflicts section for `nvidia390-dkms`, `nvidia470-dkms` and `nvidia-dkms` subpackages would be possible but I'm unsure if that's the correct way to implement this.

I'm also unsure how to implement the dependency:
Currently `nvidia-open-dkms` lists `nvidia` as a dependency, while `nvidia-dkms` does not, as that's a subpackage, thus I'm unsure what happens if one tries to install `nvidia-dkms` without `nvidia` already installed and whether it'd behave the same way with an external template with circular dependency.

As I mentioned above, I am also unsure whether it'd be better to add the open dkms module as a subpackage, and what do do with the license being set to nvidia proprietary in that package. Another thing that bugs me about the licenses is that it theoretically puts this template into the main repo while the nvidia one is in nonfree one.

Making people manually choose the module could be a possibility, I suppose, but it'd be inconvenient as heck. We could probably automate it to some degree via querying which device id are enabled unless explicitly called for (for now the readme provided by the upstream lists the pci device ids of supported devices) — though that's me fantasizing the possibilities, I have little to no idea whether xbps is capable of operating on that level.

JkktBkkt commented:
I must have overlooked that section, thanks for pointing it out to me

Made this [patch](https://pastebin.com/t6EPs1AW) (to be applied when both nvidia-open-dkms and nvidia are up-to-date, i.e. this PR locally merged)

Renamed nvidia-dkms dependency into nvidia-proprietary-dkms, which provides new virtual dependency: 
nvidia-meta-dkms, also provided by nvidia-open-dkms (default)
Removed "replaces=" section from nvidia-open-dkms in favor of "provides=" and "conflicts="

The installation order and specifically [dkms pkg](https://github.com/void-linux/void-packages/blob/73bb93844902d2266243b9768538c663985a97d0/srcpkgs/nvidia/template#L255) section in nvidia template are not adjusted here, I am wondering if messing with that order by installing the nvidia-open-dkms contents way later may break anything.

JkktBkkt commented:
Am testing 570.144 locally since we haven't merged the proprietary version yet.
Had to change the distfiles url to nvidia's site because release is still N/A on gh

JkktBkkt commented:
Tried to rename the branch, that closed the PR, then restored, local git refused to push into it, so I tried to recreate the branch, but then couldn't re-open due to force-pushing. Sorry for the spam.

Tyfuzzle commented:
Hi, any ideas when this might get merged in?

JkktBkkt commented:
> Hi, any ideas when this might get merged in?

Unfortunately this isn't ready for merging. 
I'm essentially bumping versions, running daily with this (it installs and runs fine but upgrading is a bit of a nuisance, check the template file for instructions) and am waiting for feedback/guidance from maintainers.

At the very least, there is still an open question on how to handle the versions and the choice between open kernel modules vs proprietary ones (open support the newest cards but don't support older ones. proprietary supports older cards but not the newest).

fvalasiad commented:
^

But I'd suspect the answer is soon this will get attention. Nvidia themselves will now treat the open kernel modules as the default, it's only natural to expect distros to transition, and that doesn't exclude void.

Use this build if you want it now, otherwise be patient.

Tyfuzzle commented:
I went ahead and installed it about 5 days ago. Kinda needed it as I upgraded to a 5080. Haven't had any problems so far.

JkktBkkt commented:
Added the "conflicts" field, but "replaces" is kept for ease of installation for testing

vikigenius commented:
@Tyfuzzle I am using 5070 TI and can't get the nvidia drivers to work properly. Is this necessary for newer cards? Do I need to set the nvidia-drm.modeset=1 in the kernel params still?

Tyfuzzle commented:
@vikigenius I only followed the instructions on the template. Didn't need to do anything else. As far as I'm aware, the closed source DKMS module doesn't support newer cards. Haven't ever used any kernel parameters. If you're skeptical that it might be breaking something, remove the parameter. If it doesn't change anything, put it back.

vikigenius commented:
@Tyfuzzle thanks I installed this package but I am getting issues with running niri

> None of the following EGL extensions is supported by the underlying EGL implementation at least one is required EGL_EXT_device_query

I created an issue here https://github.com/YaLTeR/niri/issues/2095

But I am not sure, if the error could be because of this package? Do you use wayland or xorg?

vikigenius commented:
@JkktBkkt thanks a lot I did manage to get it working reasonably well.

I do run into this issue intermittently https://bbs.archlinux.org/viewtopic.php?pid=2254844#p2254844

Random crashes for apparently no reason at all. The entire system freezes up, I can VT switch to a different TTY but the login there hangs there as well. It's as if the OS itself crashes

JkktBkkt commented:
@vikigenius can't really say whether that is actually caused by the nvidia-open driver. There is some hope that the same fix that was applied in the 580 driver branch to fix the vkps/llama.cpp issue you've mentioned made it into the 570.181 release as well, but I haven't tested it. 
The other user also mentioned disabling integrated graphics (in UEFI) in the thread you've linked, perhaps that would work?

Regarding updates: 
- Minor kernel changes get handled automatically: when you install a new kernel version through xbps, hooks for kernels get run — building DKMS modules, meaning Dynamic Kernel Module Support (like this one), get rebuilt automatically, kernel can get signed, initramfs for it gets generated, etc. Similar processes exist for removing kernel versions, for clean-up.
- Major kernel versions can (but not necessarily will) introduce incompatibilities. The same hooks will still run but it can result in a module that doesn't build or builds and is misbehaving in one way or another. In that case you'd need to have a compatible version of the driver installed and/or roll back to kernel version that is compatible with the one available.
- Driver updates need to be handled manually in the current state. See the contents of the template for an outline of what needs to be done.

Tyfuzzle commented:
> @Tyfuzzle thanks I installed this package but I am getting issues with running niri a wayland compositor
> 
> > None of the following EGL extensions is supported by the underlying EGL implementation at least one is required EGL_EXT_device_query
> 
> I created an issue here [YaLTeR/niri#2095](https://github.com/YaLTeR/niri/issues/2095)
> 
> But I am not sure, if the error could be because of this package? Do you use wayland or xorg?

Sorry for not responding sooner. Not that it matters because you mentioned youve got it working now. I'm using x11 at the moment.

I've ran into my own problem though now. The main nvidia package has upgraded to 580 but this PR is still on 570 so now I'm stuck in terminal with xinit crashing (or I assume I am because of incompatibility between the driver and kdms module version).

JkktBkkt commented:
> upgraded to 580 but this PR is still on 570

@Tyfuzzle glad you figured this out. 

For anyone else in a similar situation, here's the gist of it:
I've tested 580.76.05 open locally today and had an issue with all gtk4 applications that I've tested simply hanging (as mentioned in the accepted PR for 580)
Not sure if it only applies to the open kernel modules, but I think it's best to keep 570.181 for now. 
You can use `xdowngrade` to downgrade the package and then hold it to prevent upgrading for the time being.
See here: https://docs.voidlinux.org/xbps/advanced-usage.html

In case you've removed the older versions from your local cache or didn't have them, the easiest would probably be to: check out this PR as-is to a branch, not rebasing it on top of master, then packaging `nvidia` locally, and then using the same xdowngrade command but pointing to your local repos: 
`void-packages/hostdir/binpkgs/current-branch-name/nonfree/`
and `void-packages/hostdir/binpkgs/current-branch-name/multilib/nonfree` if you use 32bit libraries too
Install `nvidia-570.181.x86_64.xbps nvidia-dkms-570.181.x86_64.xbps nvidia-firmware-570.181.x86_64.xbps nvidia-gtklibs-570.181.x86_64.xbps nvidia-libs-570.181.x86_64.xbps nvidia-opencl-570.181.x86_64.xbps` (possibly nvidia-libs-32bit-570.181.x86_64.xbps)
And then reinstalling the nvidia-open-dkms package

JkktBkkt commented:
> DId you notice this?

Sorry, I forgot to add a patch that fixes this into the commit (was present in local system), should work fine now.

On the process: I'd do 3 before 2, and 4 happens automatically when you upgrade `nvidia`, because `nvidia-dkms` is a direct dependency of `nvidia`

JkktBkkt commented:
While I have been using this on a "daily" machine ever since the original PR was created, as-was, I don't think the PR as a whole is in a ready-to-merge state here (permalink to it is in first comment)
With nvidia and nvidia-open-dkms installed and a new version available for one or both, manual intervention from the user is required to update (or temporarily holdback), which is far from ideal.

So I think a few changes for the nvidia template would help, something like the commit below works (from neither installed, it defaults to nonfree version). Not sure if this is the proper approach to take though and about details, i.e. names for virtual and the whole handling of -dkms subpackage.

I've briefly (game via proton, mpv via vulkan playback, both in a wayland session) tested updating locally from both 580.82.05 proprietary and open versions (open as it was at the time in this PR), correctly installs the required one in working state.

abenson commented:
Updating them in sync isn't that big of a deal, I don't think. We already do this with other packages.

JkktBkkt commented:
Dropped the virtual by changing the provides (quite neat that it can resolve like this) and restored the copied nouveau blacklist logic as pointed out by abby and ahesford over on irc.

vikigenius commented:
What's the difference between this PR and this https://github.com/void-linux/void-packages/pull/56685 ?

I have tried both and I find it easier to update the drivers with the other one. But maybe it becomes easier once we actually merge this?

abenson commented:
The other switches entirely to the open drivers, breaking support for pre-Turing GPUs (10xx and earlier). This PR provides the option to have both.

fvalasiad commented:
> The other switches entirely to the open drivers, breaking support for pre-Turing GPUs (10xx and earlier). This PR provides the option to have both.

While I originally did agree with this path, The announcement of pre-turing cards being deprecated post 580 driver version was what led me into opening that PR.

Is there any point in having both, knowing that? Pre-turing cards can just keep using the original template, but I don't see the reason why any of the newer ones should be defaulting to the proprietary modules.

And in case we do want to keep both for whatever reason(as far as I understand nvidia will eventually drop em completely, given that they don't support their newer series), wouldn't it make more sense for it to be the other way around? Aka by default the package installs the open ones and users can install the second package to switch to proprietary if they wish? 

Am I missing something?

fvalasiad commented:
> Or are you advocating we just drop support for pre-Turing cards altogether? We could drop `nvidia470` and `nvidia390`, as well, or we'd just have a weird gap in support.

I'm advocating for the pre Turing cards to get the same treatment the older deprecated cards got, aka:

As soon as next production release drops, e.g 590, fork the nvidia template at that point in time (being in the latest 580.x version) to a package named 'nvidia580', point users with pre-turing cards to it in the handbook.

As soon as that happens, switch the nvidia package to the open kernel modules, seeing as now only Turing and later cards will be using the nvidia package, which not only are supported by the open kernel drivers, but it's actually the recommendation by nvidia. In fact, in the case of the 5xxx series it's not only a recommendation but the only path forward. 

My understanding so far is that we do not need to keep the proprietary option around for Turing+ cards, but if there are valid reasons to, such as those stated by @JkktBkkt, then we either delay the whole switch, or go about deciding the default, which I advocate should still be the open kernel drivers, while offering a package similar to this one that switches to the proprietary as a backup.

Regardless this would result in a single step install for either group with an optional second step as a backup if deemed necessary.

Thoughts?

save-buffer commented:
Hi! I just tried to use this PR, but was hitting an issue with the DKMS not installing properly. It said it was building, but I noticed that the module never ended up in `lsmod`. After some hunting I realized that the DKMS was actually failing to be built (log here: `/var/lib/dkms/nvidia-open-dkms/580.105.08/build/make.log`) due it not being able to find the `xz` command. Installing `xz` manually made it so that I could `dkms install nvidia-open-dkms/580.105.08` manually, and now I can once again use my graphics environment! 

Should we add `xz` as a dependency to this package so that the kernel module can be built?

JkktBkkt commented:
@revington looks like the kernel panic'd a bit over 11 minutes after starting, I'd suggest looking at the rest of the logs around the 670 mark (which is seconds from boot) mark to see the rest of the message and perhaps figure out what's causing that. 
My guess would be that system is trying to go to sleep (or wake up) and the module is crashing then?

Also worth double-checking that the rest of the nvidia- packages have been updated to the same 580.119.02 version

There are reports of 580.119.02 (and 590.4x versions) being broken in various setups as well, so if you'd prefer to not figure it out much with it much, a rollback to 580.105.08 (`xdowngrade path/to/older.xbps` should help with that) or a test of the nvidia-dkms (proprietary kernel module) could be attempted to get your system back to running fine for the time being.

JkktBkkt commented:
@revington I'm not sure if xbps-query checks that all dependencies are satisfied, but if those installed match what's listed in run_depends, that should be good.
Oh, also, if you're additionally using 32-bit libs (i.e. for Steam) you want to install matching version of nvidia-libs-32bit from `hostdir/binpkgs/git_branch_name/multilib/nonfree/`

revington commented:
@JkktBkkt thanks,  finally I got it working with 580.119.02.
rebuilding dependencies worked.

halfbit0 commented:
Right now for me, the module fails to build for kernel 6.19, but builds for 6.18 and lower

[make.log](https://github.com/user-attachments/files/25503721/make.log)

JkktBkkt commented:
rev2 so that xbps doesn't try to replace `nvidia-open-dkms` with `nvidia-dkms` due to forced `nvidia` rebuild

JkktBkkt commented:
There isn't really much to do here. Nvidia released 580.142 ~~, but Void skipped it~~. Next up is the merge of #59718 which will update to 595 branch and make "open" kernel modules the default ones, so this will no longer be needed.

telamon commented:
For those of you who are stuck in limbo; this is how I restore open nvidia dkms after new driver updates using this PR.

```bash
xbps-query nvidia # grab the new version
vi src-pkgs/nvidia-open-dkms/template # bump version

xgensum -i nvidia-open-dkms # regen checksum
./xbps-src pkg nvidia-open-dkms # build package

# install it
sudo xbps-install -R hostdir/binpkgs nvidia-open-dkms

sudo reboot # all done
```

edit: @classabbyamp thanks, updated!

classabbyamp commented:
`xgensum -i` to avoid having to copy/paste the new checksum

JkktBkkt commented:
Open modules are now the default (nvidia version 595.58.03 has been merged).
